### PR TITLE
TTL enhancement

### PR DESCRIPTION
**Pull request by shiyuhang0**
### What problem does this PR solve? 
1. TTL: Fix the  bug that it won't be closed correctly after commitPrimaryKey. not retry in some cases: such as deadlock
2. Reduce the `persist` in Delete to save memory, and it won't take too much extra time according to the benchmark
3. Add Benchmark in README

### CODE REVIEW

**Comment by shiyuhang0** [comment]
PTAL @xuanyu66

**Comment by shiyuhang0** [comment]
PTAL @birdstorm

**Review by birdstorm — APPROVED** [review]
LGTM

File: tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java [hunk 1]
```diff
@@ -569,7 +569,7 @@ private boolean isTxnHeartBeatSuccess(TxnHeartBeatResponse resp)
     }
 
     if (resp.hasError()) {
-      throw new TiClientInternalException("TxnHeartBeat fail, " + resp.getError().getAbort());
+      throw new KeyException(resp.getError(), "TxnHeartBeat fail, " + resp.getError().getAbort());
```
Comment (xuanyu66) [inline tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java:572]: Plz describe in PR why change it to KeyException

**Review by xuanyu66 — APPROVED** [review]

**Pull request merged**